ols/user_data_collection/data_collector.py

```diff
@@ -0,0 +1,245 @@
+"""Collect insights and upload it to the Ingress service.
+
+It collects the data on startup and then it waits for the specified interval.
+"""
+
+import base64
+import io
+import json
+import logging
+import os
+import pathlib
+import subprocess
+import sys
+import tarfile
+import time
+
+import requests
+
+OLS_USER_DATA_PATH = os.environ.get("OLS_USER_DATA_PATH", "./user-data/")
+
+OLS_USER_DATA_COLLECTION_INTERVAL = int(
+    os.environ.get("UDC_COLLECTION_INTERVAL", 2 * 60 * 60)
+)  # 2 hours in seconds
+
+# available ingress urls are:
+# - https://console.redhat.com/ - production - works with refresh token
+# - https://console.stage.redhat.com/ - staging/dev - doesn't work with refresh token
+INGRESS_URL = os.environ.get("INGRESS_URL", "https://console.redhat.com/")
+INGRESS_ENDPOINT = os.environ.get("INGRESS_ENDPOINT", "api/ingress/v1/upload")
+INGRESS_TIMEOUT = 10  # seconds
+
+# customer portal offline token is generated from https://access.redhat.com/management/api
+CP_OFFLINE_TOKEN = os.environ.get("CP_OFFLINE_TOKEN")
```

Comment:
elsewhere you indicate this is "mainly for debug purposes", if that is 100% true then this is ok as an env var.

but if there's any expectation customers will be setting this, we need to add it to the ols config(as a filepath reference, like our other cred/secret refs) and read it from there.  meaning we'd also need the OLSConfig CRD to include a secret ref and then the ols operator to mount the secret into the upload container in the ols pod.

Reply:
Customers don't need to touch it. Developers can. This is for testing purposes.